### PR TITLE
Add typed commands

### LINKED ISSUE #3: Typed commands
Commands that we can send to agent to control it based on strings and strings joins. E.g.

```python
agent_host.sendCommand("turn -0.5")
agent_host.sendCommand("move " + str(speed))
agent_host.sendCommand("jump 1")
```

There are two suggestions:

1. Create discriminated union represented commands, like:

```fsharp
type Command =
  | Turn of float
  | Move of float
  | Jump of bool
  // etc

agentHost.sendCommand(Move speed)
```

It's rather low-level API in original Malmo commands style.

2. Add set of extensions to `AgentHost`. 

```fsharp
agentHost
  .SendTurn(-0.5)
  .SendMove(speed)
  // etc
```

It's API in fluent style.

### PR DESCRIPTION
**Pull request by gsomix**
Closes #3 

- [x] ContinuousMovementCommand
- [x] AbsoluteMovementCommand
- [x] DiscreteMovementCommand
- [x] InventoryCommand
- [x] ChatCommand
- [x] SimpleCraftCommand
- [x] MissionQuitCommand

### CODE REVIEW

File: FSharp.Malmo/AbsoluteMovementCommands.fs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿namespace FSharp.Malmo
+
+module rec AbsoluteMovementCommands =
+    type AbosluteMovementsCommand =
```
Comment (ForNeVeR) [inline FSharp.Malmo/AbsoluteMovementCommands.fs]: Why `Movements` and not `Movement`?

File: FSharp.Malmo/Commands.fs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    type DirectionX = Forwards | Backwards
+    type DirectionY = Up | Down
+    type DirectionZ = Left | Right
+
+    type ICommand = 
+        abstract Text : string
+
+    type AgentHost with
+    member this.sendCommand(command : ICommand) =
```
Comment (ForNeVeR) [inline FSharp.Malmo/Commands.fs]: I think you need to add an indent level here.

File: FSharp.Malmo/ContinuousMovementCommands.fs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        | Pitch s -> sprintf "pitch %f" s
+        | Turn s -> sprintf "turn %f" s
+        | Jump Start -> sprintf "jump 1"
+        | Jump Stop -> sprintf "jump 0"
+        | Crouch Start -> sprintf "crouch 1"
+        | Crouch Stop -> sprintf "crouch 0"
+        | Attack Start -> sprintf "attack 1"
+        | Attack Stop -> sprintf "attack 0"
+        | Use Start -> sprintf "use 1"
+        | Use Stop -> sprintf "use 0"
```
Comment (ForNeVeR) [inline FSharp.Malmo/ContinuousMovementCommands.fs]: All this `Start`/`Stop` matching seems redundant. Maybe define another function to convert `Start` into `"1"` and `Stop` into `"0"`?

File: FSharp.Malmo/DiscreteMovementCommands.fs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        | Move Forwards -> sprintf "move 1"
+        | Move Backwards -> sprintf "move -1"
+        | JumpMove Forwards -> sprintf "jumpmove 1"
+        | JumpMove Backwards -> sprintf "jumpmove -1"
+        | Strafe Left -> sprintf "strafe -1"
+        | Strafe Right -> sprintf "strafe 1"
+        | JumpStrafe Left -> sprintf "jumpstrafe -1"
+        | JumpStrafe Right -> sprintf "jumpstrafe 1"
+        | Turn Left -> sprintf "turn -1"
+        | Turn Right -> sprintf "turn 1"
```
Comment (ForNeVeR) [inline FSharp.Malmo/DiscreteMovementCommands.fs]: The same about `Left` / `Right` and so on: maybe convert them to strings in another set of functions?

File: FSharp.Malmo/InventoryCommands.fs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿namespace FSharp.Malmo
+
+module rec InventoryCommands =
+    type InventoryCommand =
+        | SwapInventoryItems of int*int
```
Comment (ForNeVeR) [inline FSharp.Malmo/InventoryCommands.fs]: You was using spaces around `*` in `AbsoluteMovementCommands.fs`, but not use them here. Please, choose a consistent style!

(I prefer spaces around `*` for that matter.)

File: FSharp.Malmo/InventoryCommands.fs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        | Hotbar of int*HotbarSwitch
+        interface ICommand with
+            member this.Text = commandToText this
+
+    let internal commandToText = function
+        | SwapInventoryItems (x, y) -> sprintf "swapInventoryItems %d %d" x y
+        | CombineInventoryItems (x, y) -> sprintf "combineInventoryItems %d %d" x y
+        | DiscardCurrentItem -> sprintf "discardCurrentItem"
+        | Hotbar (barInd, On) -> sprintf "hotbar.%d 1" barInd
+        | Hotbar (barInd, Off) -> sprintf "hotbar.%d 0" barInd
```
Comment (ForNeVeR) [inline FSharp.Malmo/InventoryCommands.fs]: <Usual comment about `On`/`Off`.>

File: FSharp.Malmo/SimpleCraftCommands.fs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿namespace FSharp.Malmo
+
+module rec SimpleCraftCommands =
+    type SimpleCraftCommand =
+        | Craft of string
+        | CraftVariation of string*string
```
Comment (ForNeVeR) [inline FSharp.Malmo/SimpleCraftCommands.fs]: I'd add names for the tuple components here. E.g. `CraftVariation of itemType : string * variation : string`.

File: FSharp.Malmo/ContinuousMovementCommands.fs (hunk 3 above)
Comment (gsomix) [inline FSharp.Malmo/ContinuousMovementCommands.fs]: OK, it's tracked as different issue.

https://github.com/gsomix/fsharp-malmo-examples/issues/10

File: FSharp.Malmo/AbsoluteMovementCommands.fs [hunk 8]
```diff
@@ -0,0 +1,20 @@
+﻿namespace FSharp.Malmo
+
+module rec AbsoluteMovementCommands =
+    type AbosoluteMovementCommand =
```
Comment (ForNeVeR) [inline FSharp.Malmo/AbsoluteMovementCommands.fs]: Hey, that aint good enough yet. `Abosolute` 😭

**Pull request merged**